### DISCUSSION TITLE
SEO product Optimization : delete categorie name

### DISCUSSION THREAD

**Discussion by jeckyl** [PrestaShop/PrestaShop#37437] [Ideas]
### Prerequisites

- [x] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [x] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Hi,
PrestaShop has always had the bad practice of embedding category names in product URLs.
`{category:/}{id}{-:id_product_attribute}-{rewrite}.html`

### Expected behavior

It would be preferable, when installing PrestaShop, for the rewrite to be without the categories :
`{id}{-:id_product_attribute}-{rewrite}.html`

Provide a redirection line from URL anchors to new ones to avoid problems when updating.

### Steps to reproduce

SEO&URLs Tab

### PrestaShop version(s) where the bug happened

1.0.0 - 9.0.0

### Your company or customer's name goes here (if applicable).

Mediacom87

**Comment by venditdevs** [comment]
Why would you want that? companies like Google want a clean, readable URL. A  product URL with categories in it is not per definition bad. It is only bad when you make a trash of the URL so Google "does not understand" what's behind the URL.

We have webshops with a complete parent/child category URL structure and they are diong very, very well on SEO base. We also have webshops with no categories in the URL, so, for example domainname.com/product-name which are also doing great.

The structure of the URL is negligible, leave the structure of the URL a decision of the end user. 

Considering it as an option would be fine, but not standard.

**Comment by jeckyl** [comment]
Simply in cases where a product is moved from one category to another by default, or when the simplified URL of a category changes, as this crashes the product URLs miserably.

**Comment by venditdevs** [comment]
That's not true, as long as there's a product ID in the product section of the URL itself, it will not crash. It wil only result in a 404 if the category changes and the URL does not contain the necessary ID's.

**Comment by Multipage** [comment]
@jeckyl  what @venditdevs says is true. There is absolute no way this causes any problems when something changes. As long as there is a product ID in the URL Prestashop will find it and redirect the 'wrong' page to the 'correct' page.

The categoryname is needed. The product can be in multiple categories. Prestashop needs to know where  the user came from (breadcrumbs)  otherwise we get really unpredicted and unrealiable behaviour of Prestashop and thus bad user experience!

This change would bring 0 to SEO optimization.

**Comment by Hlavtox** [comment]
I agree, the category in the name is useless.

### Arguments
- Shorter URL, better URL - for readability, for QR codes, for storage.
- When you change default category of a product, URL changes. Yes, there is 301, but still, it harms SEO. It's enough that I have search console full of redirects because of EAN changes, that are in the URL.
- Prestashop doesn't need it to know where the user came from, it detects it from `$_SERVER['HTTP_REFERER']`.

So, for me, for **NEW INSTALLS**, it's a yes.

Ping @kpodemski @ShaiMagal

**Comment by kpodemski** [comment]
Typical SEO issue. One is in favor of it, one is against it.

I don't see any value in this change, and please don't treat it as a "bug" or call it "bad practice" without showing sources and evidence that it actually is.

**Reply by kpodemski** [reply]
I moved it to the Discussions because **it is not a bug report**

**Reply by matks** [reply]
> Typical SEO issue. One is in favor of it, one is against it.
> 
> I don't see any value in this change, and please don't treat it as a "bug" or call it "bad practice" without showing sources and evidence that it actually is.

Wise words 😄

**Comment by jeckyl** [comment]
1. URL stability
2. Complete uselessness of URL category or form for SEO purposes
3. Ease of adding the option rather than having to go to the trouble of correcting it after the fact

**Reply by kpodemski** [reply]
p1. You say URL stability, but I've noticed that changing the product name is more often done than changing the product default category. Both will be redirected with 301 to the new URL, correct?
p2. in almost every audit I've seen so far, there was a suggestion to change from `/clothes/1-product` to `/women/clothes/1-product`, so, not only to have a category but a category path - I don't have any reason to think that those agencies weren't competent - do you have any articles that could prove your point? :-)
p3. I think it's similar to p1

**Reply by Hlavtox** [reply]
1. Yes, it will be redirected. The only required thing is product ID.
2. That doesn't mean it's correct. So many competent agencies using Prestashop, yet we discover after X years that canonicals are missing and other major SEO problems.

https://developers.google.com/search/docs/crawling-indexing/url-structure#resolution

> Whenever possible, shorten URLs by trimming unnecessary parameters.

**Reply by kpodemski** [reply]
I checked a few fashion stores, and they use product URLs without a category.

It probably doesn't impact whether there is a category or not, but given that big fashion brands decided to go this way, and I remembered that they did have categories in URLs before, maybe there's something to it.

Also, it will still be configurable so... :) It should be fine for all the scenarios.

so, eventually, @jeckyl @Hlavtox you convinced me, thanks 👌🏻 😄

**Reply by jeckyl** [reply]
In fact, if you change the product name often, you're already repeating the 301, and if you have the category in the URL and change it, the effect becomes exponential.
What's more, since the category doesn't include its exact URL with ID, you can't use the URL to access the corresponding category page, and this leads to frustration on the part of some users.

**Comment by kpodemski** [comment]
@jeckyl 

I think the next step should be to submit the pull request against the 9.0.x branch, so we don't need to wait too long for it.

I agree with @Hlavtox that this should be a default only for the new installations.

On my side, I would make sure to communicate about the change where needed and we could also emphasize that everything in this matter is still configurable.

Having a PR could allow everyone to perform the tests, etc.

Of course, this change has to also be approved by Ilies, he may have some extra questions but seeing the upvotes and comments here I'm optimistic.

Thank you for having a constructive discussion everyone.

**Reply by jeckyl** [reply]
My proposal obviously only concerns PrestaShop installation, not upgrades or backward compatibility. This modification won't change anything for older PrestaShop users, only for new installations.
Remember that adding the category after the fact is simply a matter of modifying the URL structure, whereas if you wish to remove the category from the URL, you'll need to modify the structure and the htaccess to manage the disappearance of this element.

**Comment by jeckyl** [comment]
PR ready : https://github.com/PrestaShop/PrestaShop/pull/37467

**Comment by Maofree** [comment]
Hi
I also suggest to move at the end the IDs like so
{rewrite}{-:ean13}-{id}{-:id_product_attribute}.html

the same for category
{rewrite}-{id}
for supplier
supplier/{rewrite}-{id}
and brand
brand/{rewrite}-{id}

**Reply by jeckyl** [reply]
Your proposal has nothing to do with the initial discussion. Please feel free to create a new discussion to propose this evolution and perhaps have it validated in a future version.

**Reply by Maofree** [reply]
Hi
I just created a new discussion

**Reply by jeckyl** [reply]
The discussion : https://github.com/PrestaShop/PrestaShop/discussions/37481

**Comment by kpodemski** [comment]
Thank you everyone, the PR is now merged, so I think we can close the discussion 👌🏻
